### PR TITLE
Refactor fairness metrics toolbox for classification

### PR DESCRIPTION
**Pull request by iamksuresh**
## Description

1. main.py as the main entry point
2. moved test assests to stock-plugins root (stock-plugins/user_defined_files)
3. refactored the folder and files structure.
4. Users can now execute the plugin as a module
5. Plugin will now accept comma separated assets paths (data,model,ground_truth), model classifications, ground truth value

## Type of Change

refactor: Code refactoring without changes in functionality

perf: Performance improvements 

## How to Test

Execute the below commands to run unit tests -
```
cd aiverify/stock-plugins/aiverify.stock.fairness_metrics_toolbox_for_classification/algorithms/fairness_metrics_toolbox_for_classification/
pytest  . 

```

image

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x]  I have tested the changes locally and verified that they work as expected.
- [x]  I have added or updated the necessary documentation (README, API docs, etc.).
- [x]  I have added appropriate unit tests or functional tests for the changes made.
- [x]  I have followed the project's coding conventions and style guidelines.
- [x]  I have rebased my branch onto the latest commit of the main branch.
- [x]  I have squashed or reorganized my commits into logical units.
- [x]  I have added any necessary dependencies or packages to the project's build configuration.
- [x]  I have performed a self-review of my own code.
- [x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

### CODE REVIEW

File: stock-plugins/aiverify.stock.fairness-metrics-toolbox-for-classification/algorithms/fairness_metrics_toolbox_for_classification/pytest.ini [hunk 1]
```diff
@@ -1,5 +1,4 @@
 # pytest.ini
 [pytest]
-# this is needed to exclude plugin_test.py, which is not a unit test file
-python_files = test_*.py
+testpath=tests
```
Comment (timlrx) [inline stock-plugins/aiverify.stock.fairness-metrics-toolbox-for-classification/algorithms/fairness_metrics_toolbox_for_classification/pytest.ini:3]: I get the following warning message when running pytest:
```
PytestConfigWarning: Unknown config option: testpath
```

I think it should by `testpaths`.

**Review by timlrx — COMMENTED** [review]
Tested that all the test cases pass (other than the warning errors) and that the built package works with the example shell script in the readme

File: stock-plugins/aiverify.stock.fairness-metrics-toolbox-for-classification/algorithms/fairness_metrics_toolbox_for_classification/tests/e2e/test_e2e.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    [
+        multiclass_pipeline,
+        tabular_binary_pipeline,
+        multiclass_non_pipeline,
+        binary_non_pipeline,
+        image_pipeline,
+    ],
+)
+def test_fairness_metrics_toolbox_for_classification_plugin(data_set):
+    try:
```
Comment (timlrx) [inline stock-plugins/aiverify.stock.fairness-metrics-toolbox-for-classification/algorithms/fairness_metrics_toolbox_for_classification/tests/e2e/test_e2e.py:97]: I don't think the test is running correctly due to the try, except block!

**Comment by iamksuresh** [comment]
@imda-peckyoke  : The image_pipeline data set is failing with the below error.So, I am commenting it for now.
```
(There was an error deserializing the model: XXX/aiverify/stock-plugins/aiverify.stock.fairness-metrics-toolbox-for-classification/algorithms/fairness_metrics_toolbox_for_classification/../../../user_defined_files/pipeline/bc_image_face)
```
@imda-kelvinkok  : can you help check if the data set is valid ?

ref - https://github.com/aiverify-foundation/aiverify/pull/313/files#diff-55cdd7bb8b61c78e907d7e4cf5a58b741221edfb870a0e933fb66a514034dcc5R93

**Review by imda-peckyoke — APPROVED** [review]
Approved

**Pull request merged**